enable/brush.py

```diff
@@ -0,0 +1,224 @@
+        """
+        return array([self.offset] + list(self.color_))
+
+    @observe("+update")
+    def observe_update_traits(self, event):
+        self.updated = True
+
+
+class Gradient(HasStrictTraits):
+    """ A color gradient. """
+
+    #: The sequence of color stops for the gradient.
+    stops = List(Instance(ColorStop), update=True)
+
+    #: A trait which fires when the gradient is updated.
+    updated = Event()
+
+    def to_array(self):
+        """ Return a sorted list of stop arrays.
+
+        This is the raw form of the stops required by Kiva.
+
+        Returns
+        -------
+        stop_array_list : arrays
+            A list of array of (offset, r, b, b, a) values corresponding to
+            the color stops.
+        """
+        return array([
+            stop.to_array()
+            for stop in sorted(self.stops, key=attrgetter("offset"))
+        ])
+
+    @observe("stops.items.updated")
+    def observe_stops(self, event):
+        self.updated = True
+
+    def _stops_default(self):
+        return [
+            ColorStop(offset=0.0, color="white"),
+            ColorStop(offset=1.0, color="black"),
+        ]
+
+
+class GradientBrush(Brush):
+    """ A brush that paints a color gradient. """
+
+    #: The sequence of color stops for the gradient.
+    gradient = Instance(Gradient, args=(), allow_none=False, update=True)
+
+    #: How the gradient extends beyond the main area of the gradient.
+    #:
+    #: pad
+    #:    Extend using the first and last colors.
+    #: reflect
+    #:    Extend using a reflection of the gradient.
+    #: repeat
+    #:    Extend by repeating the gradient.
+    spread_method = Enum("pad", "reflect", "repeat", update=True)
+
+    #: The coordinate space used by the gradient points.
+    #:
+    #: userSpaceOnUse
+    #:     Coordinates are specified in the current graph context's coordinate
+    #:     system.
+    #: objectBoundingBox
+    #:     Coordinates are specified between 0.0 and 1.0 and are relative to
+    #:     the bounding box of the path being filled.
+    units = Enum("userSpaceOnUse", "objectBoundingBox", update=True)
+
+    @observe("gradient:updated")
+    def observe_stops(self, event):
+        self.updated = True
+
+
+class LinearGradientBrush(GradientBrush):
+    """ A brush that paints a linear color gradient. """
+
+    #: The start point of the linear gradient.
+    start = Tuple(Float, Float, update=True)
+
+    #: The stop point of the linear gradient.
+    end = Tuple(Float, Float, update=True)
+
+    def set_brush(self, gc):
+        """ Apply the brush settings to a graphics context.
+
+        This calls linear_gradient on the GC with the appropriate values.
+
+        Parameters
+        ----------
+        gc : graphics context
+            The graphics context to use the brush with.
+        """
+        gc.linear_gradient(
+            *self.start,
+            *self.end,
+            self.gradient.to_array(),
```

Comment:
What do you think about caching this array, so that it doesn't have to be built every time the gradient is used?